### DISCUSSION TITLE
Feedback: Change search.minChars default to 2

### DISCUSSION THREAD

**Discussion by geoffreymcgill** [retypeapp/retype#206] [General]
Hello Retypers!

We have been working with the search component functionality recently and uncovered a small change that we would like to make, but before making the change we would like your feedback.

Currently, the search results are returned once `3` characters have been entered in the search box. We would like to change the default to `2` characters.

There are a several reasons why we are proposing this change:

1. Searching for two-character strings in software are common, such as `UI`. With a three char minimum, those strings are not easily found.
2. Returning results when two chars are entered is faster. The search component is already super fast, so results are returned in a snap. We don't see a need to wait for three chars.

No changes within your project are required. After upgrading to the next Retype release (`v1.11`), your site would just start working automatically with the new value and return search results when two characters are entered.

Since day 1, the value has also been configurable using the [`search.minChars`](https://retype.com/configuration/project/#minchars) config. For instance, returning results when a single character is supplied would require the following configuration in your projects `retype.yml` file:

```yml
search:
  minChars: 1
```

We would like to include this change in the upcoming Retype `v1.11` release.

If you have an opinion on this, please let us know. 👍 or 👎

**Comment by DannyBen** [comment]
So the change in essence is:

- Default value to 2 instead of 3
- People who have this configured will not be affected
- People who hate it can configure it back to 3

Right? If so, the question is what is the downside?

A couple of downsides that I can think of, that may or may not exist:

1. The resulting list will be too long, creating a more awkward user experience
2. The resulting list will include less relevant things at the top

**Comment by geoffreymcgill** [comment]
> So the change in essence is:
> 
> - Default value to 2 instead of 3
> - People who have this configured will not be affected
> - People who hate it can configure it back to 3

You got it! Nice terse summary.

As well, stop words such as `in`, `on`, `at`, `he`, etc are still applicable and would not be returned in the search results.

---

> 1. The resulting list will be too long, creating a more awkward user experience

Currently, only the top 20 results are returned, but of course that is also configurable using the [`search.maxResults`](https://retype.com/configuration/project/#maxresults) project config. Revising to `search.minChars: 2` would not create a longer list of results.

> 2. The resulting list will include less relevant things at the top

Correct, mostly. The results are still ranked based on relevancy score, but of course 3 chars provides more context than 2 chars. Initial results with 2 chars will undoubtedly be less relevant, but just keep typing. If you are searching for `ui`, the search component will give a higher score for `Configure UI` then `Set a guid`, so it does weight exact word matches higher.

**Comment by geoffreymcgill** [comment]
[`search.minChars: 2`](https://retype.com/configuration/project/#minchars) is now the default as of [`v1.11.0`](https://github.com/retypeapp/retype/releases/tag/v1.11.0). 

Thanks for the feedback and hope this helps.
